client/src/partials/purchase/view/purchase_records.js

```diff
@@ -1,24 +1,21 @@
-angular.module('bhima.controllers')
-.controller('purchaseRecords', [
-  '$scope',
-  '$q',
-  'connect',
-  function ($scope, $q, connect) {
-    var session = $scope.session = { purchase_type: 'indirect'};
-    $scope.purchase_filter = {};
+
+var purchaseRecords = function ($q, connect) {
+  var that = this;
```

Comment:
You are rigth, that is good but less meanning full for me me according to the angular context, I will use vm.